src/twisted/words/test/test_jabberxmlstream.py

```diff
@@ -21,12 +21,13 @@
 from twisted.words.protocols.jabber import error, ijabber, jid, xmlstream
 
 try:
-    from twisted.internet import ssl
+    from twisted.internet import ssl as _ssl
 except ImportError:
     ssl = None
-    skipWhenNoSSL = "SSL not available"
+    skipWhenNoSSL = True
 else:
-    skipWhenNoSSL = None
+    ssl = _ssl
+    skipWhenNoSSL = False
```

Comment:
To reduce code duplication, maybe have something like this:

```
    ssl = None
    skipWhenNoSSL = (True, "SSL not available")
else:
    ssl = _ssl
    skipWhenNoSSL = (False, "")

@skipIf(*skipWhenNoSSL)
def test_wantedSupported(self):
```

Now I know that is a lot of work, so maybe just leave the duplicate code as it is.

Reply:
And since there are a lot of SSL skip conditions, maybe it would make sense to have a shared code like this

```
from twisted.trial import conditionals


@conditionals.skipWhenNoSSL
def test_wantedSupported(self):
```

In this way, all the SSL skipping code is controller from a single place